### PR TITLE
New pair style ylz

### PR DESCRIPTION
**Pull request by mehdibghk**
**Summary**

The ylz pair-style computes anisotropic interactions between pairs of particles considering the relative particle orientations usable to model lipid bi-layer membranes through one-particle-thick coarse-grained particles to capture the mechanical properties, such as gel-fluid-gas phase transition lipids, diffusion, and bending rigidity. 

**Author(s)**
Mehdi Baghaee, postdoctoral researcher at department of mechanical and aerospace engineering, Southern University of Science and Technology (SUSTech), Shenzhen, China.
Email address: mehdi.bgh@gmail.com

**Licensing**
By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Implementation Notes**
A new pair-style (ylz) is implemented through adding new source code file (pair_ylz.cpp) and a header file (pair_ylz.h) to ASPHERE package of source (src) folder. Two relevant examples (flat_membrane and vesicle) are provided to the ASPHERE package of example folder. Documentation of the pair-style are addressed in rst file (doc/src/pair_ylz.rst).
The pair-style was verified through several research works such as:
Yuan H, Huang C, Li J, Lykotrafitis G, Zhang S, "One-particle-thick, solvent-free, coarse-grained model for biological and biomimetic fluid membranes", Physical review. E, Statistical, Nonlinear, and Soft Matter Physics, 12 Jul 2010, 82, 011905.
Fu SP, Peng Z, Yuan H, Kfoury R,  c, Young YN, "Lennard-Jones type pair-potential method for coarse-grained lipid bilayer membrane simulations in LAMMPS", Computer Physics Communications 210 (2017) 193–203. 
Becton M, Averett RD, Wang X, "Flow plate separation of cells based on elastic properties: a computational study", DOI: 10.1007/s10237-018-1093-9.

### CODE REVIEW

**Comment by akohlmey** [comment]
@mehdibghk thanks for your contribution. I have make some changes to better align with the LAMMPS programming conventions and style. Also applied some corrections to the documentation and added a test to ensure that all particles are ellipsoids (otherwise there would be segfaults or memory corruption).

**Review by stanmoore1 — APPROVED** [review]

**Comment by mehdibghk** [comment]
@akohlmey thanks for your productive comments and corrections. I also enriched the content of the rst file with more information such as explanations, figures and references.

**Comment by akohlmey** [comment]
> @akohlmey thanks for your productive comments and corrections. I also enriched the content of the rst file with more information such as explanations, figures and references.

That is a bit overkill for a "technical" documentation. Please do not use GIF animations, anyway. They are not supported by PDFLaTeX. We like the documentation to be concise and to the point and it should be convertible to HTML, PDF, and e-book formats.

If you want to highlight your research and applications of this pair style, please consider creating a short writeup and provide the animations with it, so we can add it to the LAMMPS homepage at: https://www.lammps.org/movies.html

File: doc/src/pair_ylz.rst [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+was originally developed as a particle-based solvent-free model for biological membranes
+:ref:`(Yuan2010a) <Yuan>`. Unlike the :doc:`pair_style gayberne <pair_gayberne>` whose 
+orientation dependence is strictly derived from the closest distance between two 
+ellipsoidal rigid bodies, the orientation-dependence of this pair style is mathematically
+defined such that the particles can self-assemble into one-particle-thick 
+fluid membranes (see the movies below).
+
+.. only:: html
+
+.. image:: JPG/self_assemble_into_membranes.gif
```
Comment (akohlmey) [inline doc/src/pair_ylz.rst]: Not only are those GIF animations not supported by PDF, they also are **far** too large to be part of the LAMMPS distribution. So please undo the recent changes to the documentation (if you are experienced in git, please try a hard reset to the point before adding the images and then a forced push and afterwards re-apply only doc changes that are unrelated to the animations. Those can be added to the LAMMPS homepage as already noted).

**Review by akohlmey — CHANGES_REQUESTED** [review]

**Review by akohlmey — APPROVED** [review]

**Review by sjplimp — APPROVED** [review]

**Comment by sjplimp** [comment]
@mehdibghk Thanks for contributing this new pair style.  Would it be difficult (at some point, not now) to add the capability to model mixtures of ellipsoidal particles and spheres (or points).  E.g. for a solvated membrane.  Similar to what pair gayberne allows for a mixture of ellipsoidal and LJ particles.  In GB it is just some logic to treat the non-ellipsoidal particles in a simpler way, with no orientation dependence.  And 2 non-ellipsoids collapse to the LJ formula.  If your YLZ potential just treated the mixing
case, then pure LJ interactions could be assigned via pair hybrid.

**Pull request merged**